src/kspace.cpp

```diff
@@ -224,12 +232,12 @@ void KSpace::ev_setup(int eflag, int vflag, int alloc)
   evflag = 1;
 
   eflag_either = eflag;
-  eflag_global = eflag % 2;
-  eflag_atom = eflag / 2;
+  eflag_global = eflag & ENERGY_GLOBAL;
+  eflag_atom = eflag & ENERGY_ATOM;
 
   vflag_either = vflag;
-  vflag_global = vflag % 4;
-  vflag_atom = vflag / 4;
+  vflag_global = vflag & (VIRIAL_PAIR | VIRIAL_FDOTR);
+  vflag_atom = vflag / 4; // TODO
```

Comment:
@sjplimp if `vflag = VIRIAL_CENTROID` (=8), this will be `2` and therefore true. Not sure if that is intentional or an outdated code path.

Reply:
This is intentional. It would have been cleaner to have written `vflag_atom = vflag & 12;`
The centroid description of the per-atom stress is equivalent to the virial description for all interactions with spherical symmetry (pair styles like lj/cut or morse). Thus all such pair styles are automatically compatible and that also includes KSpace. Only bonded interactions and many-body potentials need a different derivation of the per-atom stress (for global stress the result is the same again for both representations).

Reply:
@akohlmey thanks for the clarification.